sdk/src/gateway/client.ts

```diff
@@ -405,6 +408,52 @@ export class GatewayApiClient {
         return tokens.map((token) => ADDRESS_LOOKUP[this.chainId][token]).filter((token) => token !== undefined);
     }
 
+    /**
+     * Builds the parameters required to stake ERC-20 tokens using the specified strategy.
+     *
+     * @param stakeParams - The parameters required for staking.
+     * @returns {Promise<StakeERC20Params>} The constructed staking parameters.
+     * @throws {Error} If the strategy or token does not match, or if any address is invalid.
+     *
+     * @note Tokens must be approved first before calling the staking function.
+     */
+    async buildStake(stakeParams: BuildStakeParams): Promise<StakeTransactionParams> {
+        const strategies = await this.getStrategies();
+
+        // Convert addresses to lowercase and check if strategy exists
+        const strategy = strategies.find((s) => s.address.toLowerCase() === stakeParams.strategyAddress.toLowerCase());
```

Comment:
import { isAddressEqual } from 'viem';
...
```suggestion
        const strategy = strategies.find((s) => isAddressEqual(s.address, stakeParams.strategyAddress));
```